### ISSUE TITLE
GitGraph integration in Power BI

### ISSUE THREAD

**Issue by stchalon** [nicoespeon/gitgraph.js#359]
Hello
I discovered gitGraph a couple of days ago using ParcelJS. I could create my first graph and would like to have the graph integrated into a Power BI report. I can generate an image that I import as a file but I am wondering if this could be possible to integrate directly the script. I could not find a document explaining how to make it

I would like to have a possibility from a R script to launch the gitgraph.js that would generate a HTML page that I could display in a report thanks to radacad guide

Alternative, not that nice, is to create an image from ParcelJS display and import in a static PBI report

Radacad: https://radacad.com/interactive-map-using-r-and-power-bi-create-custom-visual-part-1

**Comment by fabien0102** [comment]
Hello 👋 
I never played with R or PowerBI, but if you can generate HTML, you can definitely use gitgraph to generate the graph ;) No need for ParcelJS here!

The simple way (without parceljs) is on the README (https://github.com/nicoespeon/gitgraph.js/tree/master/packages/gitgraph-js)

```html
<head>
  <!-- Load the JS file -->
  <script src="https://cdn.jsdelivr.net/npm/@gitgraph/js"></script>
</head>
```
Just take the version for jsdeliver (or host the file yourself) and you can directly use gitgraph :wink:

The challenge is to map the powerBI data to fit the gitgraph api, I would suggest to generate a compatible json with R and inject it with `gitgraph.import`

https://github.com/nicoespeon/gitgraph.js/blob/46a98c9f145224d0c41afb472b5580bc4644aaa1/packages/stories/src/gitgraph-js/4-imports.stories.tsx#L17

Have fun (and keep us up-to-date, I'm curious about this integration ^^)

**Comment by stchalon** [comment]
Thanks Fabien. Indeed I will try to shortcut ParcelJS and found Power BI Visual HTML Viewer that I will try. I will keep you posted if I succeed ;-) and share a kind of HowTo if some of your users are interested 

Stéphane 

Envoyé de mon iPhone

**Comment by stchalon** [comment]
Hi Fabien,
Coming back to you to give you an update on where I stand in the
integration. I tried several tracks, one of them sounds good but I fear
this is with GitHub v1 only, using GitGraphRef.html below

0. As reference, I used GitGraphRef.html that I can display from a
classical browser
1. Using R script launched from Power BI. Example below that is *working*
in Power BI Desktop, not on Shiny, *but *displayed out of Power BI in a new
tab of my browser. not what I wanted to get

library(pander)

openFileInOS("c:/tmp/GitGraph.html")

2. Using HTML Viewer Visual in Power BI. This is working on Power BI
Desktop but no more when I publish my .pbi on Shiny. The other issue is
that Visual 'HTML Viewer' has been discontinued by provider e.g. not a
long term solution
3. D3.js is a Visual able to display charts in a PBI iFrame but not a good
track as this would be too intrusive for GitGraph and more, don't really
know how to make this ;-)
4. Easiest one working quite well but, as I wrote, I could use a basic
.html not the example from GitGraph v2 example, no idea why

In Power BI, Create a Dashboard : New -> Dashboard
[image: image.png]

Add a Tile, Web content, Paste GitGraphRef.html in code area
[image: image.png]


Apply. You are done!
[image: image.png]

Where I am a bit frustrated is that I would have preferred to have a
GitGraph using metro template is to be aligned on GitGraph v2 *if* the
GitGraphRef.html is expressed as I believe in GitGraph v1. not very sure...


Kind regards,
Stéphane


Le dim. 24 mai 2020 à 15:56, Fabien BERNARD <notifications@github.com> a
écrit :
